numba/core/base.py

```diff
@@ -789,7 +789,7 @@ def printf(self, builder, format_string, *args):
             cstr = self.insert_const_string(mod, format_string)
         else:
             cstr = format_string
-        fnty = Type.function(Type.int(), (GENERIC_POINTER,), var_arg=True)
+        fnty = llvmir.FunctionType(llvmir.IntType(32), (GENERIC_POINTER,), var_arg=True)
```

Comment:
Suggest adding a test like:
```diff
diff --git a/numba/tests/test_cgutils.py b/numba/tests/test_cgutils.py
index dba8108..8832050 100644
--- a/numba/tests/test_cgutils.py
+++ b/numba/tests/test_cgutils.py
@@ -9,7 +9,7 @@ import numpy as np
 import unittest
 from numba.core import types, typing, cgutils, cpu
 from numba.core.compiler_lock import global_compiler_lock
-from numba.tests.support import TestCase
+from numba.tests.support import TestCase, run_in_subprocess
 
 
 machine_int = ir.IntType(types.intp.bitwidth)
@@ -117,5 +117,35 @@ class StructureTestCase(TestCase):
             inst.b = ir.Constant(ir.FloatType(), 4.56)
 
 
+class TestCGContext(TestCase):
+    """Tests for code generation context functionality"""
+
+    def test_printf(self):
+        # Tests the printf() method
+
+        value = 123456
+
+        code = f"""if 1:
+        from numba import njit, types
+        from numba.extending import intrinsic
+
+        @intrinsic
+        def printf(tyctx, int_arg):
+            sig = types.void(int_arg)
+            def codegen(cgctx, builder, sig, llargs):
+                cgctx.printf(builder, \"%d\\n\", *llargs)
+            return sig, codegen
+
+        @njit
+        def foo():
+            printf({value})
+
+        foo()
+        """
+
+        out, _ = run_in_subprocess(code)
+        self.assertIn(str(value), out.decode())
+
+
 if __name__ == '__main__':
     unittest.main()
```

Reply:
This was added in 2b0a05943cc17bf1780a0e7f4a53d84dbe6ae592 -- thank you @njriasan